Add to migration doc steps regards changes over metrics to the scaffolded files. (#2625)
operator-framework/operator-sdk: doc/migration/version-upgrade-guide.md

## Unreleased

### Bug Fixes and Improvements for Metrics

> Some changes were performed in the default implementation regards the default metrics export.  In this way, update the  `cmd/main.go` file as follows. 

Reviewer: rephrase. "There were some changes to the default implementation of the metrics export. These changes require the `cmd/main.go` be updated as follows."

> Some changes were performed in the default implementation regards the default metrics export.  In this way, update the  `cmd/main.go` file as follows. 

Author: ```suggestion
There were some changes to the default implementation of the metrics export. These changes require the `cmd/main.go` be updated as follows.
```

		}
	}
}

// serveCRMetrics gets the Operator/CustomResource GVKs and generates metrics based on those types.
// It serves those metrics on "http://metricsHost:operatorMetricsPort".
func serveCRMetrics(cfg *rest.Config, operatorNs string) error {
	// The function below returns a list of filtered operator/CR specific GVKs. For more control, override the GVK list below
	// with your own custom logic. Note that if you are adding third party API schemas, probably you will need to
	// customize this implementation to avoid permissions issues.
	filteredGVK, err := k8sutil.GetGVKsFromAddToScheme(apis.AddToScheme)
	if err != nil {
		return err
	}

	// The metrics will be generated from the namespaces which are returned here.
	// NOTE that passing nil or an empty list of namespaces in GenerateAndServeCRMetrics will result in an error.
	ns, err := kubemetrics.GetNamespacesForMetrics(operatorNs)
	if err != nil {
		return err
	}

	// Generate and serve custom resource specific metrics.
	err = kubemetrics.GenerateAndServeCRMetrics(cfg, ns, filteredGVK, metricsHost, operatorMetricsPort)
	if err != nil {
		return err
	}
	return nil
}
```

> **NOTE**: For further and detailed information check the PRs which are responsible for the above changes [#2606](https://github.com/operator-framework/operator-sdk/pull/2606),[#2603](https://github.com/operator-framework/operator-sdk/pull/2603) and [#2601](https://github.com/operator-framework/operator-sdk/pull/2601).

Reviewer: rephrase a bit, "For more information check the PRs ...."

> **NOTE**: For further and detailed information check the PRs which are responsible for the above changes [#2606](https://github.com/operator-framework/operator-sdk/pull/2606),[#2603](https://github.com/operator-framework/operator-sdk/pull/2603) and [#2601](https://github.com/operator-framework/operator-sdk/pull/2601).

Author: ```suggestion
**NOTE**: For more information check the PRs which are responsible for the above changes [#2606](https://github.com/operator-framework/operator-sdk/pull/2606),[#2603](https://github.com/operator-framework/operator-sdk/pull/2603) and [#2601](https://github.com/operator-framework/operator-sdk/pull/2601).
```
